marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/MarathonRunTask.kt

```diff
@@ -38,17 +39,15 @@ private val log = MarathonLogging.logger {}
 open class MarathonRunTask : DefaultTask(), VerificationTask {
     @Input
     lateinit var flavorName: String
-    @Input
+    @Internal
     lateinit var applicationVariant: BaseVariant
-    @Input
+    @Internal
     lateinit var testVariant: TestVariant
-    @Input
-    lateinit var extensionConfig: MarathonExtension
     @InputDirectory
     lateinit var sdk: File
-    @Input
-    lateinit var cnf: Configuration
-    @Input
+    @Internal
+    lateinit var extensionConfig: MarathonExtension
+    @Internal
     lateinit var exceptionsTracker: ExceptionsReporter
     @Input
```

Comment:
The same about that `Input`
You should rethink what fields make sense to be input because now it could be hard for clients to realize why their tasks are UP-TO-DATE

Reply:
Caching is actually disabled with outputs.upToDateWhen { false }